docs/cluster-deployment-guide.md

```diff
@@ -365,6 +365,10 @@ If a database schema upgrade is required, it will be called out directly in the
 Some releases require changes to the schema, and some do not.
 We ensure that any consecutive versions are compatible in terms of database schema upgrades, features, and system behavior, however there is no guarantee that there is compatibility between _any_ 2 non-consecutive versions.
 
+When upgrading your Temporal Server version, ensure that you upgrade sequentially. For example, when upgrading from v1.n.x, always upgrade to v1.n+1.x (or the next available version) and so on until you get to the required version.
+Check the [Temporal Server releases](https://github.com/temporalio/temporal/releases) and follow these releases in order (you can skip patch versions).
```

Comment:
Shall we mention that no need to go through all patch version, and we recommend to use latest patch of the minor version.

Reply:
It would also be great to briefly mention why we require the server to be upgrade through all minor versions. 
The major reason is server only guarantee code backward compatibility between 2 consecutive minor versions. Keep all old code around forever is not practical. For example, shard metadata handling code would need to handle both old shard data and double write to support new format. That old code will eventually be removed from code base so skip version might cause the old format become unrecognizable.